pipeline/hard-reset.sh

```diff
@@ -0,0 +1,263 @@
+   exit 1
+fi
+logdir="logs/${net_profile}"
+msg "profile: ${net_profile}"
+msg "execute: ${cmd}"
+if [ "${force_yes}" = false ] ;then
+  printf "[Y]/n > "
+  read -r yn
+  if [ "${yn}" != "Y" ] ;then
+     exit
+  fi
+fi
+
+######### functions ###########
+function hard_reset_shard
+{
+   shard=$1
+   msg "cat $logdir/shard${shard}.txt | xargs -i{} -P50 bash -c \"./restart_node.sh -U -X -D -y -d logs/${net_profile} -p ${net_profile} -t 0 -r 0 -R 0 {}\""
+}
+
+function check_consensus
+{
+   shard=$1
+   msg "./run_on_shard.sh -p ${net_profile} -T $shard 'tac /home/tmp_log/*/zeorlog*.log | grep -m 1 HOORAY'"
+}
+
+function check_version
+{
+   shard=$1
+   msg "./run_on_shard.sh -p ${net_profile} -T $shard './harmony -version'"
+}
+
+function restart_watchdog
+{
+   NET=$(echo $net_profile | tr [a-z] [A-Z])
+   msg "./update_watchdog.sh -w $NET -t $net_profile -u -p -r -y"
+   ./update_watchdog.sh -w $NET -t $net_profile -u -p -r -y
+}
+
+function _do_hard_reset_per_shard
+{
+   shard=$1
+   hard_reset_shard ${shard}
+   check_consensus ${shard}
+   check_version ${shard}
+}
+
+function do_jenkins_release
+{
+   msg 'TODO: curl jenkins command'
+   msg 'ex: https://jenkins.harmony.one/job/harmony-release/413/api/json'
+}
+
+function restart_sentry
+{
+   msg "restart sentry"
+#   KEYDIR=${HSSH_KEY_DIR:-~/.ssh/keys}
+   for num in $(seq 1 ${configs[benchmark.shards]}); do
+      if [ "${configs[sentry${num}.enable]}" == "true" ]; then
+         sentryip=${configs[sentry${num}.ip]}
+         sentryshard=${configs[sentry${num}.shard]}
+         sentryuser=${configs[sentry${num}.user]}
+         msg "sentry${num}: $sentryuser@$sentryip/$sentryshard"
+#         key=$(find_key_from_ip $sentryip)
+# TODO: testing
+         echo ./node_ssh.sh ${sentryuser}@${sentryip} <<EOC
+         tmux new -s sentry -d || { tmux kill-session -t sentry && tmux new -s sentry -d; } ;
+         sleep 1 ;
+         tmux send-keys -t sentry "./auto_node.sh run --shard $sentryshard --auto-active --clean --network ${configs[benchmark.network_type]} --beacon-endpoint https://api.s0.${configs[flow.rpczone]}.hmny.io" Enter;
+EOC
+      fi
+   done
+}
+
+function restart_explorer
+{
+   msg "restart explorer backend"
+   explorerip=$(host ${configs[explorer.name]} | awk ' { print $NF } ')
+# TODO: testing
+   echo ./node_ssh.sh ${explorerip} <<EOC
+      tmux new -s explorerbe -d || { tmux kill-session -t explorerbe && tmux new -s explorerbe -d; } ;
+      sleep 1 ;
+      tmux send-keys -t explorerbe "npm run db-reset; sleep 3; npm run start" Enter;
+EOC
+      fi
+}
+
+function restart_dashboard
+{
+   msg "clear dashboard db"
+   dashboardip=$(host ${configs[dashboard.name]} | awk ' { print $NF } ')
+# TODO: testing
+   echo ./node_ssh.sh ${dashboardip} <<EOC
+      tmux new -s dashboardbe -d || { tmux kill-session -t dashboardbe && tmux new -s dashboardbe -d; } ;
+      sleep 1 ;
+      tmux send-keys -t dashboardbe "cd appengine && npm install" Enter;
```

Comment:
this is incorrect.

Similar to my above comment on block explorer command, let me add npm commands we can use in this script for auto-recovery.
Please update here as:
`cd frontend; npm run db-reset-confirm; cd ../appengine; ...` (... will be cmds for deploying appengine. @john-harmony can fill this in)

Reply:
command: please update this to:
```cd frontend; npm run db-reset -- -n [ostn|pstn|stn] (-y); cd ..```